### PR TITLE
[GPU] Add shape agnostic optimized FullyConnectedIMAD kernel

### PR DESCRIPTION
**Pull request by andrew-k-park**
### Details:
 - Shape agnostic optimized fully_connected_gpu_imad kernel implementation

### Tickets:
 - 104931

### CODE REVIEW

**Review by ahnyoung-paul — APPROVED** [review]

File: src/plugins/intel_gpu/src/kernel_selector/cl_kernels/fully_connected_gpu_imad.cl [hunk 1]
```diff
@@ -36,19 +37,30 @@ KERNEL(fully_connected_gpu_imad)(
...
-    const uint batch = (uint)get_global_id(1) * TILE_BATCH;
+    const uint batch = (uint)get_global_id(1) * tile_batch;
     const uint skip_f = (uint)get_global_id(2);
 #endif
 
     // Accumulators initialization
     MAKE_VECTOR_TYPE(int, TILE_OFM) dotProd[TILE_BATCH];
     MAKE_VECTOR_TYPE(uint, TILE_OFM) idx_w;
-    unroll_for (uint ob_idx = 0; ob_idx < TILE_BATCH; ob_idx++) {
+    unroll_for (uint ob_idx = 0; ob_idx < tile_batch; ob_idx++) {
```
Comment (yeonbok) [inline src/plugins/intel_gpu/src/kernel_selector/cl_kernels/fully_connected_gpu_imad.cl:63]: For shape agnostic kernel, tile_batch is not constant, so please do not "unroll_for".

**Comment by yeonbok** [comment]
LGTM if perf check is done, + minor comments are resolved.

File: src/plugins/intel_gpu/src/kernel_selector/kernels/fully_connected/fully_connected_kernel_imad.cpp [hunk 2]
```diff
@@ -86,14 +89,21 @@ bool FullyConnectedKernelIMAD::Validate(const Params& params, const optional_par
     const auto& fc_params = static_cast<const fully_connected_params&>(params);
     const auto& in = fc_params.inputs[0];
     const auto& wei = fc_params.weights;
+    auto out_l = fc_params.outputs[0].GetLayout();
+
+    // Dynamic kernel doesn't support dynamic weights yet
```
[the remark this answers is not in the document]
Comment (andrew-k-park) [reply src/plugins/intel_gpu/src/kernel_selector/kernels/fully_connected/fully_connected_kernel_imad.cpp]: removed it

File: src/plugins/intel_gpu/src/kernel_selector/cl_kernels/fully_connected_gpu_imad.cl (hunk 1 above)
Comment (andrew-k-park) [reply src/plugins/intel_gpu/src/kernel_selector/cl_kernels/fully_connected_gpu_imad.cl:66]: update to do not loop unrolling for shape agnostic kernel

**Review by yeonbok — APPROVED** [review]

**Comment by andrew-k-park** [comment]
There is no perf regression for static case

**Pull request merged**